pages/api/id/[form]/bearer.ts

```diff
@@ -0,0 +1,42 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { getSession } from "next-auth/client";
+import * as Constants from "../../../../lib/constant";
+import { executeQuery } from "../../../../lib/integration/queryManager";
+
+export const retrieve = async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+  try {
+    const session = await getSession({ req });
+    // Only allow a valid user session
+    if (!session || session === null || req.method !== "GET")
+      return res.status(403).json({ error: "Access Denied" });
+    await getToken(req.query.form as string);
+  } catch (err) {
+    res.status(500).json({ error: `Error on Server Side ${err}` });
+  }
+  /**
+   * Will return a bearer token if there is one associated with a given formID
+   * it might return a Not found (404) if token doest exist ( undefined or null)
+   * otherwise 400 as status code
+   * @param formID
+   * @returns
+   */
+  async function getToken(formID: string) {
+    type BearerResponse = {
```

Comment:
Create a type for the Template record as a whole. Also this should be in a separate file so it could be leveraged elsewhere. This file seems like a good place https://github.com/cds-snc/platform-forms-client/blob/main/lib/types.tsx